Desktop/Tempest/Providers/Network/NetworkConnectionProvider.cs

```diff
@@ -405,18 +405,32 @@ private void Accept (object sender, SocketAsyncEventArgs e)
 
 			if (!this.running || e.SocketError != SocketError.Success)
 			{
-				if (!this.running)
-					e.Dispose();
-				else
+				try
 				{
-					#if !SILVERLIGHT
-					e.AcceptSocket.Shutdown (SocketShutdown.Both);
-					e.AcceptSocket.Disconnect (true);
-					//    ReliableSockets.Push (this.reliableSocket);
-					#else
-					e.ConnectSocket.Dispose();
-					#endif
-					BeginAccepting (e);
+					if (!this.running)
+						e.Dispose();
+					else
+					{
+						#if !SILVERLIGHT
+						e.AcceptSocket.Shutdown (SocketShutdown.Both);
+						e.AcceptSocket.Disconnect (true);
+						//    ReliableSockets.Push (this.reliableSocket);
+						#else
+						e.ConnectSocket.Dispose();
+						#endif
+					}
+				}
+				catch (Exception ex)
```

Comment:
Well the specific exception I was seeing was a SocketException, but is there really any value in letting any exception escape here?